### PR TITLE
Feature/xaptc 32

### PR DESCRIPTION
**Pull request by ColmCooney**
Protractor tests(Not yet tested with UI)

### CODE REVIEW

File: ui-src/e2e/pageObjects/configurationPageObject.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    rampDelay = element(by.name('rampDelay'));
+    testCaseDir =element(by.name('testCaseDir'));
+    testSuiteDir =element(by.name('testSuiteDir'));
+    baseStatsDir = element(by.name('baseStatsOutputDir'));
+    reportsDir = element(by.name('reportOutputDir'));
+    
+    label = $('json-schema-form').$$('label');
+    required = element(by.className('help-block'));
+   
+    configFileLocation = "../../../config/"
```
Comment (ConorSmyth) [inline ui-src/e2e/pageObjects/configurationPageObject.ts]: This should be a const

File: ui-src/e2e/components/configurationComponent.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,436 @@
+import { browser, element, by, By, $, $$, ExpectedConditions, protractor, Key } from "protractor"
+import { read } from "fs";
```
Comment (ConorSmyth) [inline ui-src/e2e/components/configurationComponent.e2e-spec.ts]: I dont see this used.

File: ui-src/e2e/pageObjects/configurationPageObject.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
+import { browser, element, by, By, $, $$, ExpectedConditions } from "protractor"
+import { read } from "fs";
```
Comment (ConorSmyth) [inline ui-src/e2e/pageObjects/configurationPageObject.ts]: Not sure where this is used?

File: ui-src/package.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "automated-perf-test",
+  "version": "0.0.0",
+  "license": "MIT",
+  "scripts": {
+    "ng": "ng",
+    "build": "ng build --prod",
+    "watch": "ng build --watch",
+    "test": "ng test",
+    "lint": "ng lint",
```
Comment (ConorSmyth) [inline ui-src/package.json]: @angerooney11  & @ColmCooney  `ng lint` Can you run this and fix the issues that arise, it should go a long way to fixing most of the issues.

File: ui-src/src/app/automated-ui-service.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+};
+
+@Injectable()
+export class AutomatedUIService {
+  constructor(private http: HttpClient) {}
+
+  private url = "http://localhost:9191/configs";
+
+  addConfig(form: Data): void {
+    console.log("Form", form);
```
Comment (ConorSmyth) [inline ui-src/src/app/automated-ui-service.ts]: Angela and I went through these your 2 commits behind @ 16h11 16Jan

File: ui-src/src/app/test-cases/test-cases.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { async, ComponentFixture, TestBed } from "@angular/core/testing";
+
+import { TestCasesComponent } from "./test-cases.component";
+
+describe("TestCasesComponent", () => {
+  let component: TestCasesComponent;
+  let fixture: ComponentFixture<TestCasesComponent>;
+
+  beforeEach(
+    async(() => {
```
Comment (ConorSmyth) [inline ui-src/src/app/test-cases/test-cases.component.spec.ts]: @ColmCooney  & @angerooney11  Have we checked that the unit tests run. Cause async afaik is not available until node 7

File: ui-src/src/styles.css [hunk 7]
```diff
@@ -0,0 +1,15 @@
+
+/* You can add global styles to this file, and also import other style files */
+@import '~https://fonts.googleapis.com/css?family=Roboto:300,400,500,700';
+@import '~https://fonts.googleapis.com/icon?family=Material+Icons';
+@import "~@angular/material/prebuilt-themes/indigo-pink.css";
```
Comment (ConorSmyth) [inline ui-src/src/styles.css]: We don't need this any more.

File: uiServices/homepage.go [hunk 8]
```diff
@@ -52,6 +53,35 @@ func getIndexPage() *chi.Mux {
 		w.Header().Set(contentTypeHeader, htmlType)
 		w.Write([]byte(htmlBytes))
 	})
+
```
Comment (ConorSmyth) [inline uiServices/homepage.go]: Check these changes with @frankquinn

File: ui-src/.angular-cli.json [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+      "main": "main.ts",
+      "polyfills": "polyfills.ts",
+      "test": "test.ts",
+      "tsconfig": "tsconfig.app.json",
+      "testTsconfig": "tsconfig.spec.json",
+      "prefix": "app",
+      "styles": [
+        "styles.css",
+        "../node_modules/bootstrap/dist/css/bootstrap.min.css",
+        "../node_modules/@angular/material/prebuilt-themes/indigo-pink.css"
```
[the remark this answers is not in the document]
Comment (angerooney11) [reply ui-src/.angular-cli.json]: this is needed for the styling with the json schema

File: .gitignore [hunk 10]
```diff
@@ -1,3 +1,50 @@
+<<<<<<< HEAD
```
Comment (jenny-flynn) [inline .gitignore]: merge conflict?

File: schema.json [hunk 11]
```diff
@@ -1,28 +1,55 @@
...
+    "apiName": { "type": "string" },
+    "targetHost": { "type": "string" },
+    "targetPort": {
+      "type": "string",
+      "minimum": 1,
+      "maximum": 65535
+    },
+    "memoryEndpoint": { "type": "string" },
+    "numIterations": { "type": "integer", "minimum": 0 },
+    "concurrentUsers": { "type": "integer", "minimum": 0 },
```
Comment (jenny-flynn) [inline schema.json]: use the requirements to build these minimum values. Anywhere it says "positive integer", use a minimum of 1. I don't know how nicely defaults work, but there are some defaults in the story and ye can come up with sensible defaults too.

**Pull request closed without merging**